localstack-core/localstack/services/apigateway/next_gen/execute_api/helpers.py

```diff
@@ -53,3 +55,36 @@ def render_uri_with_path_parameters(uri: str, path_parameters: dict[str, str]) -
         uri = uri.replace(f"{{{key}}}", value)
 
     return uri
+
+
+def get_source_arn(context: RestApiInvocationContext):
+    method = context.resource_method["httpMethod"]
+    path = context.resource["path"]
+    return (
+        "arn:aws:execute-api"
+        f":{context.region}"
+        f":{context.account_id}"
+        f":{context.api_id}"
+        f"/{context.stage}/{method}{path}"
+    )
+
+
+def get_lambda_function_arn_from_invocation_uri(uri: str) -> str:
+    """
+    "arn:aws:apigateway:us-east-1:lambda:path/2015-03-31/functions/arn:aws:lambda:us-east-1:123456789012:function:SimpleLambda4ProxyResource/invocations",
+    :param uri: the integration URI value for a lambda function
+    :return: the lambda function ARN
+    """
+    return uri.split(":lambda:path")[1].split("functions/")[1].removesuffix("/invocations")
```

Comment:
It seems like you are doing an extra split? Wouldn't splitting once at `"functions/"` be enough?